### DISCUSSION TITLE
not working for mac os 12.4

### DISCUSSION THREAD

**Discussion by rohanprofile** [kyleneideck/BackgroundMusic#603] [General]
i tried installing as per instruction in read me but the app does not work ...the volume slider only works for chrome app i haven't tested all but many ..and it works for none ..Also after installing it says this ::: 

Cleaning
[1/3] Installing the virtual audio device Background Music Device.driver to /Library/Audio/Plug-Ins/HAL
[2/3] Installing BGMXPCHelper.xpc to /usr/local/libexec
[3/3] Installing Background Music.app to /Applications
Restarting coreaudiod to load the virtual audio device.
Launching Background Music.
Done.
rm: BackgroundMusic-master/BGM.xcworkspace/xcshareddata/swiftpm: Permission denied
rm: BackgroundMusic-master/BGM.xcworkspace/xcshareddata: Permission denied
rm: BackgroundMusic-master/BGM.xcworkspace: Permission denied
rm: BackgroundMusic-master: Permission denied

I am new to mac os need help....

**Comment by EnYiHou** [comment]
I got the same issue, any update?

**Comment by kyleneideck** [comment]
I think https://github.com/kyleneideck/BackgroundMusic/wiki/App-Volumes-Fixes will explain it.

Those `rm` errors just mean that `build_and_install.sh` wasn't able to delete the `BackgroundMusic-master` folder after it finished installing. It looks like `xcodebuild` left `BackgroundMusic-master/BGM.xcworkspace/xcshareddata/swiftpm` with permissions that won't let your user delete it. You should be able to delete `BackgroundMusic-master` in Finder or with `sudo rm -r BackgroundMusic-master`.
